File: cpp/bfsdfs.cpp
#include <iostream>
using namespace std;

#define MAX 20

int graph[MAX][MAX];
int visited[MAX];
int V; 


void BFS(int start) {
    int queue[MAX];
    int front = 0, rear = 0;

    for (int i = 0; i < V; i++)
        visited[i] = 0;

    visited[start] = 1;
    queue[rear++] = start;

    cout << "BFS Traversal starting from vertex " << start << ": ";

    while (front < rear) {
        int node = queue[front++];
        cout << node << " ";

        for (int i = 0; i < V; i++) {
            if (graph[node][i] == 1 && visited[i] == 0) {
                visited[i] = 1;
                queue[rear++] = i;
            }
        }
    }
    cout << endl;
}


void DFS(int start) {
    cout << start << " ";
    visited[start] = 1;

    for (int i = 0; i < V; i++) {
        if (graph[start][i] == 1 && visited[i] == 0) {
            DFS(i);
        }
    }
}

int main() {
    int E;
    cout << "Enter number of vertices: ";
    cin >> V;
    cout << "Enter number of edges: ";
    cin >> E;

    for (int i = 0; i < V; i++)
        for (int j = 0; j < V; j++)
            graph[i][j] = 0;

    cout << "Enter edges (u v):" << endl;
    for (int i = 0; i < E; i++) {
        int u, v;
        cin >> u >> v;
        graph[u][v] = 1;
        graph[v][u] = 1; 
    }

    int start;
    cout << "Enter starting vertex: ";
    cin >> start;

    BFS(start);

    for (int i = 0; i < V; i++)
        visited[i] = 0;

    cout << "DFS Traversal starting from vertex " << start << ": ";
    DFS(start);
    cout << endl;

    return 0;
}

File: python/bfsdfs.py
def bfs(graph, start):
    visited = [False] * len(graph)
    queue = []
    labels = ["A", "B", "C", "D","E"] 
    
    visited[start] = True
    queue.append(start)
    
    while queue:
        
        vertex = queue.pop(0)
        print(labels[vertex], end=" ")
        
      
        for i in range(len(graph[vertex])):
            if graph[vertex][i] == 1 and not visited[i]:
                visited[i] = True
                queue.append(i)

def dfs(graph, start):
    visited = [False] * len(graph)
    stack = [start]
    labels = ["A", "B", "C", "D","E"]   

    while stack:
        node = stack.pop()  
        if not visited[node]:
            visited[node] = True
            print(labels[node], end=" ")

            
            for i in range(len(graph[node]) - 1, -1, -1):
                if graph[node][i] == 1 and not visited[i]:
                    stack.append(i)


n=int(input("Enter the number of vertices: "))
graph = [[0]*n for _ in range(n)]
e=int(input("Enter the number of edges: "))
print("Enter edges in the format: u v")
for i in range(e):
    u,v=map(int,input().split())
    graph[u][v]=1
    graph[v][u]=1

start=int(input("Enter the starting vertex: "))   
print("BFS starting from vertex A:")
bfs(graph, start)
print("\nDFS starting from vertex B:")
dfs(graph, start)